src/core/components/Header/Header.tsx

```diff
@@ -0,0 +1,24 @@
+import './Header.css';
+import { IconButton } from '../IconButton';
+import { BurgerMenuIcon } from '../../../assets/icons/BurgerMenuIcon';
+import { ProfileIcon } from '../../../assets/icons/ProfileIcon';
+import { useState, MouseEvent } from 'react';
+
+interface HeaderProps {
+  companyName: string;
+}
+export const Header = ({ companyName }: HeaderProps) => {
+  const [selected, setSelected] = useState('');
+
+  const handleButtonClick = (event: MouseEvent<HTMLButtonElement>) => {
```

Comment:
Let's use this function to trigger a corresponding redux action. 
In order to do that please define new action in the core slice, that doesn't change any state (simple call, or signal):
```typescript
  HEADER_BLOCK_BUTTON_CLICKED: (state, action) => {},
 ```
And export it:
`HEADER_BLOCK_BUTTON_CLICKED:  callHeaderBlockButtonClick`

The action should be imported into the current component and dispatched on a button click event. The id of button is passed as argument of the action

Reply:
Hope my "variant" is going to work.